### PR TITLE
engine: add titan key only option

### PR DESCRIPTION
**Pull request by Connor1996**
## What have you changed? (mandatory)
Cause DeleteFilesInRange may expose old blob index keys, setting key only for Titan to avoid referring to missing blob files.

## What are the type of the changes? (mandatory)
- Bug fix (change which fixes an issue)

## How has this PR been tested? (mandatory)
unit-test

## Does this PR affect documentation (docs) or release note? (mandatory)
No

## Does this PR affect tidb-ansible update? (mandatory)
No

### CODE REVIEW

**Comment by siddontang** [comment]
IMO, we should provide a Trait that can support Key only operations, not only for Titan.

**Comment by Connor1996** [comment]
@siddontang Yes, I will extract a trait later. Let's keep this and fix the issue first.

**Comment by Connor1996** [comment]
PTAL @yiwu-arbug @zhangjinpeng1987

**Comment by zhangjinpeng1987** [comment]
Build failed
```
error[E0061]: this function takes 0 parameters but 1 parameter was supplied

   --> src/raftstore/store/region_snapshot.rs:808:20

    |

808 |         let path = TempDir::new("test-titan-delete-files-in-range").unwrap();

    |                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected 0 parameters



error[E0061]: this function takes 0 parameters but 1 parameter was supplied

    --> src/raftstore/store/region_snapshot.rs:1018:20

     |

1018 |         let dir1 = TempDir::new("test-snap-cf-db-apply").unwrap();

     |                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected 0 parameters



error: aborting due to 2 previous errors

```

**Comment by Connor1996** [comment]
build fixed. PTAL again @yiwu-arbug @zhangjinpeng1987

File: src/config.rs [hunk 1]
```diff
@@ -77,7 +77,7 @@ fn memory_mb_for_cf(is_raft_db: bool, cf: &str) -> usize {
 #[serde(default)]
 #[serde(rename_all = "kebab-case")]
 pub struct TitanCfConfig {
-    pub min_blob_size: u64,
+    pub min_blob_size: ReadableSize,
```
Comment (overvenus) [inline src/config.rs]: Does it compactable with pervious config?

Comment (Connor1996) [reply src/config.rs]: Yes

**Pull request merged**